w3af/core/data/url/handlers/gzip_handler.py

```diff
@@ -57,7 +57,7 @@ def http_response(self, request, response):
                     data = gzip.GzipFile(fileobj=StringIO(body)).read()
                 elif "deflate" in enc_hdr:
                     body = response.read()
-                    data = zlib.decompress(body)
+                    data = zlib.decompress(body, -zlib.MAX_WBITS)
```

Comment:
I considered this as well. Initially I tried to set the wbits to 0, which according to the manual automatically determines the window size of the header, but for reasons unknown to me the same exception was thrown.

I have not seen any target that I could potentially test against besides the one that caused the problem. Do you know of any targets which use deflate? It would make it easier to test.

I also read somewhere that gzip could be used as a drop-in replacement of zlib in these cases, however upon further examination I discovered that gzip uses zlib. Perhaps it would still be an issue if handled by gzip.